pkg/validate/validate_test.go

```diff
@@ -871,6 +899,42 @@ func TestHierarchical(t *testing.T) {
 			},
 			wantErrs: fake.Errors(validate.InvalidSyncCode),
 		},
+		{
+			name: "RepoSync manages itself",
```

Comment:
Do we need test cases for nested RootSyncs in hierarchical mode as well?

Reply:
Nested RootSyncs are not supported in hierarchical mode because it requires declaring the `config-management-system` namespace explicitly in the path of `namespaces/config-management-system/ns.yaml`. However, cms should not be a user managed namespace.

Reply:
Ack, thanks for the explanation. I'm not very familiar with hierarchical mode